Update 02_merge_detection_data notebook with real datasets

wonjuleee opened the pull request:
<!-- Contributing guide: https://github.com/openvinotoolkit/datumaro/blob/develop/CONTRIBUTING.md -->

### Summary

<!--
Resolves #111 and #222.
Depends on #1000 (for series of dependent commits).

This PR introduces this capability to make the project better in this and that.

- Added this feature
- Removed that feature
- Fixed the problem #1234
-->

### How to test
<!-- Describe the testing procedure for reviewers, if changes are
not fully covered by unit tests or manual testing can be complicated. -->

### Checklist
<!-- Put an 'x' in all the boxes that apply -->
- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

cih9088 reviewed (changes requested):
In the second paragraph,
> "Then, we will export the merged dataset as `yolo` format."

but no `yolo` export example cell found in the notebook.

wonjuleee commented:
> In the second paragraph,
> 
> > "Then, we will export the merged dataset as `yolo` format."
> 
> but no `yolo` export example cell found in the notebook.

Correct, I updated the introduction and added filter and reindex operations.

bonhunko reviewed (approved):
LGTM

wonjuleee committed:
update merge_det_data with real datasets

wonjuleee committed:
update explanations
